File: data/models/src/commonMain/kotlin/red/razvan/restcountries/data/models/DetailedCountry.kt
// Copyright 2025 Răzvan Roșu
// SPDX-License-Identifier: Apache-2.0

package red.razvan.restcountries.data.models

data class DetailedCountry(
  val id: CountryId,
  val officialName: String,
  val commonName: String,
  val emojiFlag: String,
  val flag: Flag,
  val currencies: List<Currency>,
  val capital: List<String>,
  val continents: List<String>,
  val languages: List<Language>,
) {
  val isOfficialNameDifferentFromCommon: Boolean
    get() = !officialName.equals(commonName, ignoreCase = true)

  data class Flag(
    val png: String,
    val svg: String,
    val contentDescription: String,
  )
}

File: ios-app/rest-countries/Sources/Models/DetailedCountry.swift
//
//  DetailedCountry.swift
//  rest-countries
//
//  Created by Răzvan Roşu on 13/08/25.
//

public struct DetailedCountry: Identifiable, Hashable {
  public let id: CountryId
  public let officialName: String
  public let commonName: String
  public let emojiFlag: String
  public let flag: Flag
  public let currencies: [Currency]
  public let capital: [String]
  public let continents: [String]
  public let languages: [Language]

  public var isOfficialNameDifferentFromCommon: Bool {
    !(officialName.caseInsensitiveCompare(commonName) == .orderedSame)
  }

  public struct Flag: Hashable {
    public let png: String
    public let svg: String
    public let contentDescription: String
  }
}
